src/Server/Remote/PhpSecLib.php

```diff
@@ -102,10 +104,15 @@ public function checkConnection()
     /**
      * {@inheritdoc}
      */
-    public function run($command)
+    public function run($command, $timeout = null)
     {
+        if ($timeout === null) {
+            $timeout = self::TIMEOUT;
+        }
+
         $this->checkConnection();
 
+        $this->sftp->setTimeout($timeout);
         $result = $this->sftp->exec($command);
```

Comment:
How about to add here check that operation wasn't reached timeout like that
        
        if ($this->sftp->isTimeout()) {
            throw new \RuntimeException('Operation timeout');
        }
        
// cc @elfet , @oanhnn